bobweb/bob/message_board_service.py

```diff
@@ -1,103 +1,164 @@
                  is_chat_specific: bool = False):
-        self.starting_from = starting_from
+        self.local_starting_from = local_starting_from
         self.message_provider = message_provider
         # If scheduled message is chat specific, each message is created separately for each board and the chat id is
         # given as a parameter to the message_provider. Otherwise, the scheduled message is created only once and the
         # same result is updated to each chats message board.
         self.is_chat_specific = is_chat_specific
 
 
-def create_schedule(hour: int, minute: int, message_provider: Callable[[], Awaitable[MessageWithPreview]]):
+def create_schedule(hour: int, minute: int, message_provider: Callable[[], Awaitable[MessageWithPreview]]) \
+        -> ScheduledMessageTiming:
+    """
+    Creates schedule for message that has same content in each chat and is not chat specific in any way.
+    :param hour: staring hour in Finnish local time
+    :param minute: staring minute in Finnish local time
+    :param message_provider: async method which invocation produces the MessageWithPreview
+    :return: Scheduled message timing
+    """
     return ScheduledMessageTiming(datetime.time(hour, minute), message_provider)
 
 
-def create_schedule_with_chat_context(hour: int, minute: int,
-                                      message_provider: Callable[[MessageBoard, int], Awaitable[MessageBoardMessage]]):
+def create_schedule_with_chat_context(
+        hour: int, minute: int, message_provider: Callable[[MessageBoard, int], Awaitable[MessageBoardMessage | None]]) \
+        -> ScheduledMessageTiming:
+    """
+    Creates schedule for message that is chat specific and which content is created for each chat separately.
+    :param hour: staring hour in Finnish local time
+    :param minute: staring minute in Finnish local time
+    :param message_provider: async method which invocation produces the MessageBoardMessage
+    :return: Scheduled message timing
+    """
     return ScheduledMessageTiming(datetime.time(hour, minute), message_provider, is_chat_specific=True)
 
 
-# Default schedule for the day. Times are in UTC+-0
-default_daily_schedule = [
-    create_schedule_with_chat_context(4, 00, create_weather_scheduled_message),  # Weather
-    create_schedule(7, 00, command_sahko.create_message_with_preview),  # Electricity
-    create_schedule(13, 00, command_ruoka.create_message_board_daily_message),  # Random receipt
-    create_schedule(20, 00, good_night_wishes.create_good_night_message),  # Good night
+# Localization locale for schedules
+schedule_timezone_info = zoneinfo.ZoneInfo(bob_constants.DEFAULT_TIMEZONE)
+
+# Default schedule for the day. Note! Times are in localized Finnish Time (UTC+2 or UTC+3, depending on DST).
+# Each time new update is scheduled, it is scheduled as Finnish time as stated below.
+default_daily_schedule: list[ScheduledMessageTiming] = [
```

Comment:
So what is the idea here? That you have some debug channel that spams these all at the hours specified?

Reply:
Bot uses single message as Message Board host that is then updated based on current schedule, ongoing chat events or notifications. "MessageBoard" is shown in the pinned message preview. Bot repins the host message at every startup or with the '/Ilmoitustaulu'-command.

![image](https://github.com/user-attachments/assets/47af9c59-552a-40ec-9f97-11f0649bf0d8)

Reply:
One possible way to solve #66 and #72